src/xhr/lib/xhr.js

```diff
@@ -0,0 +1,98 @@
+import errorReporter from './errorReporter';
+import XhrError from './XhrError';
+import { configureRequestOptions, parseResponseBody } from './xhrHelpers';
+import XhrLog from './XhrLog';
+import XhrRequest from './XhrRequest';
+async function xhr(options) {
+    const baseURL = options.baseURL;
+    const path = options.path;
+    // rm any props not part of the Request/RequestInit spec
+    delete options.baseURL;
+    delete options.path;
+    // TODO: Guard against not having enough parameters
+    let err;
+    let response;
+    const xhrLog = new XhrLog();
+    return new Promise((resolve) => {
+        const xhrOptions = configureRequestOptions(options);
+        // path may be either a PathBuilder, or for convenience a path string e.g. path: 'api/posts',
+        const resolvedPath = typeof path === 'string' ? path : path.buildPath();
+        // Create a new XhrRequest object that will be passed into browser fetch
+        const request = new XhrRequest(`${baseURL}/${resolvedPath}`, xhrOptions);
+        // Begin timing how long our request will take
+        request.setTimestamp();
+        // Log request
+        xhrLog.request(request);
+        fetch(request)
+            .then((rawResponse) => {
+            return parseResponseBody(rawResponse, xhrOptions);
```

Comment:
we seem to parse the body and then see if it is "ok". I think it makes sense to check if the response was successful to begin with

Reply:
I dunno, this was Tim's code and it seems he's taking the raw response, converting it to `xhrResponse` which in TS land is backed by an interface. Then, he checks the `.ok` status. We might be able to simplify some of this in a later iteration I agree.

Reply:
Also, I think it's more useful to look at the .ts code as this is generated